### PR TITLE
2018-01-01 sync of api changes from 2017-03-01

### PR DESCRIPTION
**Pull request by solankisamir**
Sync changes in repo 2017-03-01 to 2018-01-01 
Then make 2018-01-01 version specific nit change

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by solankisamir** [comment]
@sergey-shandar can you take a look at this PR.
This is mostly about copying apis from older version to newer version.
- First commit contains the copy from the 2017-03-01
- Second commit contains changes/fixes done in 2018-01-01

**Comment by solankisamir** [comment]
@sergey-shandar a gentle ping on this.

**Comment by sergey-shandar** [comment]
@solankisamir I'm reviewing the PR.

**Comment by solankisamir** [comment]
@sergey-shandar  the CI failed for some other spec Microsoft.AzureStack/stable/2017-06-01/Registration.json. Nothing to do with this spec. Can you take a look today? I am waiting on this PR to send out PR for .NET SDK. Thanks

**Comment by sergey-shandar** [comment]
@solankisamir I will check the CI issues.

**Comment by sergey-shandar** [comment]
@ravbhatnagar new operations.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/apimapis.json [hunk 1]
```diff
@@ -124,7 +124,7 @@
             "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
           },
           {
-            "$ref": "#/parameters/ApiIdParameter"
+            "$ref": "#/parameters/ApiIdRevParameter"
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/apimapis.json]: >ApiIdRevParameter [](start = 34, length = 17)

this parameter matches the `GET` parameter for the same path. It was an issue in the old spec.

Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/apimapis.json]: >ApiIdRevParameter [](start = 34, length = 17)

`If-Match` is optional for `PUT` and only required when updating the entity.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/apimversionsets.json [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+                        }
+                    }
+                },
+                "x-ms-pageable": {
+                    "nextLinkName": "nextLink"
+                },
+                "x-ms-odata": "#/definitions/ApiVersionSetContract"
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/api-version-sets/{versionSetId}": {
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/apimversionsets.json]: >api-version-sets [](start = 132, length = 16)

We are going to change this resource type in the next version of the api. Opened issue [https://github.com/Azure/azure-rest-api-specs/issues/2570](https://github.com/Azure/azure-rest-api-specs/issues/2570)

**Comment by ravbhatnagar** [comment]
Looks good. As noted, the resourceType name will be changed in next api-version

**Comment by solankisamir** [comment]
@sergey-shandar rebased to the latest master.

**Comment by sergey-shandar** [comment]
@solankisamir I'm waiting for CI to see if there are other issues.

**Comment by sergey-shandar** [comment]
@solankisamir we aren't able to generate Go SDK from the changes. @jhendrixMSFT is investigating the issue.

**Comment by solankisamir** [comment]
@jhendrixMSFT @sergey-shandar any update on this. Is the issue with Go SDK resolved?

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**